comp/core/workloadmeta/params.go

```diff
@@ -0,0 +1,17 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package workloadmeta
+
+// Params provides the kind of agent we're instantiating workloadmeta for
+type Params struct {
+	AgentType  AgentType
```

Comment:
I think the AgentType should not be a parameter, except for the `RemoteProcessCollector`, the only exception.
The `CatalogType` might be more appropriate and it depends on a config parameter.

Reply:
But this is only really a semantics issue, right? Currently `AgentType` is indeed only used to decide the correct catalog. I can rename this, but I _think_ it's purely a bad-naming choice on my part.